admin_guide/out_of_resource_handling.adoc

```diff
@@ -0,0 +1,337 @@
+
+|Eviction Signal |Description
+
+|`*memory.available*`
+|`memory.available` = `node.status.capacity[memory]` - `node.stats.memory.workingSet`
+|===
+
+In future releases, the node will support the ability to trigger eviction
+decisions based on disk pressure. Until then, use
+xref:../admin_guide/garbage_collection.adoc#admin-guide-garbage-collection[container
+and image garbage collection].
+
+[[out-of-resource-eviction-thresholds]]
+=== Eviction Thresholds
+
+The node supports the ability to specify eviction thresholds that trigger the
+node to reclaim resources.
+
+Each threshold is of the following form:
+
+----
+<eviction_signal><operator><quantity>
+----
+
+* Valid `eviction-signal` tokens as defined in
+xref:out-of-resource-eviction-signals[Eviction Signals].
+* Valid `operator` tokens are `<`.
+* Valid `quantity` tokens must match the quantity representation used by
+Kubernetes.
+
+
+[[out-of-resource-soft-eviction-thresholds]]
+==== Soft Eviction Thresholds
+
+A soft eviction threshold pairs an eviction threshold with a required
+administrator-specified grace period. The node does not reclaim resources
+associated with the eviction signal until that grace period is exceeded. If no
+grace period is provided, the node errors on startup.
+
+In addition, if a soft eviction threshold is met, an operator can specify a
+maximum allowed pod termination grace period to use when evicting pods from the
+node. If specified, the node uses the lesser value among the
+`*pod.Spec.TerminationGracePeriodSeconds*` and the maximum-allowed grace period.
+If not specified, the node kills pods immediately with no graceful termination.
+
+To configure soft eviction thresholds, the following flags are supported:
+
+* `eviction-soft` describes a set of eviction thresholds (for example,
+`memory.available<1.5Gi`) that, if met over a corresponding grace period,
+triggers a pod eviction.
+* `eviction-soft-grace-period` describes a set of eviction grace periods (for
+example, `memory.available=1m30s`) that correspond to how long a soft eviction
+threshold must hold before triggering a pod eviction.
+* `eviction-max-pod-grace-period` describes the maximum-allowed grace period (in
+seconds) to use when terminating pods in response to a soft eviction threshold
+being met.
+
+[[out-of-resource-hard-eviction-thresholds]]
+==== Hard Eviction Thresholds
+
+A hard eviction threshold has no grace period and, if observed, the node takes
+immediate action to reclaim the associated starved resource. If a hard eviction
+threshold is met, the node kills the pod immediately with no graceful
+termination.
+
+To configure hard eviction thresholds, the following flag is supported:
+
+* `eviction-hard` describes a set of eviction thresholds (for example,
+`memory.available<1Gi`) that, if met, triggers a pod eviction.
+
+[[out-of-resource-eviction-monitoring-interval]]
+=== Eviction Monitoring Interval
+
+The node evaluates eviction thresholds per its configured housekeeping interval.
```

Comment:
Hmm, this is an error on my part.  We should just state that it monitors every 10s and the value is not modifiable.